### PR TITLE
[OmniLink] Use for each loop in getChildThing()

### PR DESCRIPTION
**Pull request by ecdye**
Updates the `getChildThing()` for simpler and much more elegant code.

Signed-off-by: Ethan Dye <mrtops03@gmail.com>

### CODE REVIEW

**Review by cpmeister — APPROVED** [review]
I like it.

File: bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/OmnilinkBridgeHandler.java [hunk 1]
```diff
@@ -568,21 +568,13 @@ public void dispose() {
...
-        }
-        if (!(theThing.isPresent())) {
-            theThing = getChildThing(THING_TYPE_UNIT, unitId);
+        Optional<Thing> theThing = Optional.empty();
+
+        for (ThingTypeUID uid : SUPPORTED_UNIT_TYPES_UIDS) {
+            theThing = getChildThing(uid, unitId);
+            if (theThing.isPresent()) {
+                return theThing;
+            }
```
Comment (fwolter) [inline bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/OmnilinkBridgeHandler.java:577]: You can go even further and do
```suggestion
        return SUPPORTED_UNIT_TYPES_UIDS.stream().map(uid -> getChildThing(uid, unitId)).flatMap(Optional::stream)
                .findFirst();
```

**Review by fwolter — CHANGES_REQUESTED** [review]
Me too!

**Comment by fwolter** [comment]
There are some formatting issues. You can fix them with `mvn spotless:apply`.

**Comment by ecdye** [comment]
> There are some formatting issues. You can fix them with `mvn spotless:apply`.

Whoops, should be fixed now.

**Review by fwolter — APPROVED** [review]
LGTM

**Pull request merged**